### PR TITLE
Select between field and freq in FDA

### LINKED ISSUE #32732: Muon frequency domain axis changes
Users want to be able to change the axis units for some plots. 

1. Frequency domain plot: to pick frequency (MHz) or Field (G) -> This seems ok if the `MuonGyromagneticRatio` is used

The concern is that if there are rounding errors multiple conversions will cause the result to drift (see the below script). However, if you use the same precision on the forward and backward transforms it does seem to be ok.

````
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from mantid.kernel import PhysicalConstants as const

def convert_to_field(workspace_name, out):
    """
    Apply the ConvertAxisByFormula algorithm to convert from MHz to Field.
    """
    alg = mantid.AlgorithmManager.create("ConvertAxisByFormula")
    alg.initialize()
    alg.setAlwaysStoreInADS(True)
    alg.setProperty("InputWorkspace", workspace_name)
    alg.setProperty("OutputWorkspace", out)
    alg.setProperty("Formula", 'x * 1. / 2.34937')#+str(const.MuonGyromagneticRatio))
    alg.setProperty("AxisTitle", 'Field')
    alg.setProperty('AxisUnits', 'Gauss')
    alg.execute()
    
def convert_to_MHz(ws, out):
    """
    Apply the ConvertAxisByFormula algorithm to convert from MHz to Field.
    """
    alg = mantid.AlgorithmManager.create("ConvertAxisByFormula")
    alg.initialize()
    alg.setAlwaysStoreInADS(True)
    alg.setProperty("InputWorkspace", ws)
    alg.setProperty("OutputWorkspace", out)
    alg.setProperty("Formula", 'x * 2.35')#+str(const.MuonGyromagneticRatio))
    alg.setProperty("AxisTitle", 'Frequency')
    alg.setProperty('AxisUnits', 'MHz')
    alg.execute()
    
# create some data with random noise

x = np.linspace(0,2000, 10000)
mu = 953.2
sigma = 27.6
rand = np.array([np.random.random() for _ in x])
y = (np.exp(-(x-mu)*(x-mu)/(0.5*sigma*sigma)))*(1.+0.2*(rand-.5))
ws = CreateWorkspace(x,y)
plt.plot(ws.readX(0),ws.readY(0), label="original")
input = ws

for j in range(0,100):
    field = "field_"+str(j)
    freq = "freq_"+str(j)
    convert_to_MHz(input, freq)    
    convert_to_field(freq, field)
    input = mtd[field]
    if j%10 == 0:
        plt.plot(input.readX(0), input.readY(0), label=str(j))
plt.legend()
plt.show()
```

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This allows users to select if they want to view their frequency data as MHz or Gauss (field). The conversion assumes a bare muon. 

**To test:**
Open frequency domain analysis
Load MUSR 62260

Go to the transform tab
Click `calculate` 
You should no have a nice plot with a range of 0 to 30.
Above the plot there is a drop down menu, change it from Frequency to Field
The plot will update (the range will now be from 0 to 1000)
Go to the fitting tab
Check the selected workspaces end with `units_Gauss`
Change the drop down back to Frequency
Check the selected workspaces end with `units_MHz`
The plot will update too

Go back to the transform tab
Change the drop down at the top of the tab from FFT to Maxent
Tick `reconstruct` in the table
Click calculate
You should no have a nice plot with a range of 0 to 30.
Above the plot there is a drop down menu, change it from Frequency to Field
The plot will update (the range will now be from 0 to 1000)
Go to the fitting tab
Check the selected workspaces end with `units_Gauss`
Change the drop down back to Frequency
Check the selected workspaces end with `units_MHz`
The plot will update too

Above the drop down menu for the Frequency/Field is another one, change it from Fit  to "Maxent Dual plot"
You will now see a bunch of counts curves and a frequency spectra
Change the data type to  the option with "Gauss"
The plot for the frequency spectra will now have an x-axis in Gauss (0 to 1000)
Check that the selected workspaces have updated in the fitting tab and end with `units_Gauss`
Change to the fit pane and check that is now field.

Changing the units on the fit pane will not change the maxent pane.

Fixes #32732. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
one of your testing steps says 
>> Tick `reconstruct` in the table

does this mean `output reconstructed data`? I don;t see a `reconstruct` option

**Comment by AnthonyLim23** [comment]
> one of your testing steps says
> 
> > > Tick `reconstruct` in the table
> 
> does this mean `output reconstructed data`? I don;t see a `reconstruct` option

yes it does

File: docs/source/release/v6.3.0/muon.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Frequency Domain Analysis
 New Features
 ############
 
+- Can now select the unit for the frequency spectra (MHz or Gauss), by picking either `Frequency` or `Field` respectfully.
```
Comment (AnthonyLim23) [inline docs/source/release/v6.3.0/muon.rst:19]:
```suggestion
- Can now select the unit for the frequency spectra (MHz or Gauss), by picking either `Frequency` or `Field` respectfully.

```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py [hunk 2]
```diff
@@ -64,54 +91,80 @@ def get_group_phase_tables(self, num_groups, instrument):
...
+        elif GAUSS in name:
+            return name.replace(GAUSS, MHz)
+        else:
+            return name
+
+    def range(self):
+        if self.x_label == FIELD:
+            return [0.0, 1000.0]
+        else:
+            return [0.0, 30.0]
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py:134]: if you do a maxent and an FFT it is then possible on the (sequential tab) to plot them both together. So I have opted to use the larger of the 2 ranges as the default value.

File: qt/python/mantidqtinterfaces/test/Muon/CMakeLists.txt [hunk 3]
```diff
@@ -127,6 +127,7 @@ set(TEST_PY_FILES
     plot_fit_panes/plot_time_fit_pane_model_test.py
     plot_fit_panes/plot_fit_pane_presenter_test.py
     plot_fit_panes/plot_freq_fit_pane_model_test.py
+    plot_fit_panes/plot_freq_fit_pane_presenter_test.py
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/test/Muon/CMakeLists.txt:130]:
```suggestion
    plot_widget/plot_freq_fit_pane_presenter_test.py
```

**Review by Harrietbrown — APPROVED** [review]
With all tests now passing this is ok to go ahead. Good job

**Pull request merged**